### ISSUE TITLE
Allow active material volume fraction to be a function of space

### ISSUE THREAD

**Issue by rtimms** [pybamm-team/PyBaMM#1119]
Currently we assume that the active material volume fraction `eps_s` is a scalar (but we allow the porosity `eps` to be a function of `x`, which in turn makes the inactive volume fraction `eps_inactive` a function of `x` by `eps_inactive = 1 - eps - eps_s`). 

After #1115 the surface area per unit volume will be computed based on particle shape for li-ion (e.g. `a = 3*eps_s/R` for spheres), and currently you only get `a(x)` if you have `R(x)`. Some of the leading-order models are only correct for constant `a` (e.g. leading order electrolyte conductivity), which is fine when `eps_s` is constant as they also only work for constant `R`. That is, it doesn't make sense to use SPM if `R=R(x)`. These will need to be updated if we allow `eps_s(x)` which lets you have `a(x)` but with constant `R`. 

You can also formulate a MPM (similar to SPM but with `R(x)`, see e.g. https://www.sciencedirect.com/science/article/pii/S0013468620302541), but the formulation is significantly different for it to require new submodels, so would be a separate issue.

**Comment by tinosulzer** [comment]
This has been fixed, presumably by #1262 but maybe earlier
